parse/IntValueRefParser.cpp

```diff
@@ -149,16 +150,25 @@ parse::int_arithmetic_rules::int_arithmetic_rules(
           ]
         ;
 
+    total_fighter_shots
```

Comment:
do you use `TotalFighterShots` in a script? If not, why is it parseable? And why does it need to be a ValueRef at all?

Reply:
That is for calculating the number of fighter shots of a carrier for use inside of the damage estimation valuerefs of hangar parts. Launch patterns are too complicated for FOCS (you would need something like a loop and variables). So this valueref can calculate the correct number of shots.

For current content it is not necessary. But if one adds a e.g. close-range fighter  (so only combatTargets = bout > 2) one needs this.

Reply:
Why does this need a subcondition parameter?

I'd have thought you could do `Source.TotalFighterShots` and get all the info needed from the object whose property is being queried...

Reply:
Note I intended to use the `TotalFighterShots` it in the missile prototype, but as i tightly control the launch pattern there (launching missiles only in bout 1) I actually know how many shots to expect without the `TotalFighterShots` valueref.

Reply:
Added a parameter `carrier = SHIP_ID` for specifying the subject of the valuerref (instead of implicitly using the source)